File: 1221-ElementAppearingMoreThan25InSortedArray/1221-ElementAppearingMoreThan25InSortedArray.java
// Last updated: 3/26/2026, 1:50:31 AM
class Solution {
    public int findSpecialInteger(int[] arr) {
        int max = 0;
        int element = -1;
        for(int i=0;i<arr.length;i++){
            int j=i+1;
            while(j<arr.length && arr[j] == arr[i]){
                j++;
            }
            if(max < (j-i)){
                max = (j-i);
                element = arr[i];
            }
            i=i+(j-i);
        }
        return element;
    }
}

File: 1221-ElementAppearingMoreThan25InSortedArray/1221-ElementAppearingMoreThan25InSortedArray.cs
// Last updated: 3/26/2026, 1:50:29 AM
public class Solution {
    public int FindSpecialInteger(int[] arr) {
        int minCount = (arr.Length/4);
        int count = 1;
        
        for(int i=1;i<arr.Length;i++)
        {
            if(arr[i]==arr[i-1])
                count++;
            else
                count=1;
            
            if(count>minCount)
                return arr[i];
        }
        
        return arr[0];
    }
}